packages/editor-ui/src/views/NodeView.vue

```diff
@@ -352,6 +375,60 @@ export default mixins(
 
 				this.$externalHooks().run('execution.open', { workflowId: data.workflowData.id, workflowName: data.workflowData.name, executionId });
 			},
+			async openWorkflowTemplate (templateId: string) {
+				this.setLoadingText('Loading template');
+				this.resetWorkspace();
+
+				let data: IWorkflowTemplate | undefined;
+				try {
+					this.$externalHooks().run('template.requested', { templateId });
+					data = await this.$store.dispatch('workflows/getWorkflowTemplate', templateId);
+
+					if (!data) {
+						throw new Error(`Workflow template with id "${templateId}" could not be found!`);
+					}
+					
+					data.workflow.nodes.forEach((node) => {
+						if (!this.$store.getters.nodeType(node.type)) {
+							const name = node.type.replace('n8n-nodes-base.', '');
+							throw new Error(`The ${name} node is not supported`);
+						}
+					});
+				} catch (error) {
+					this.$showError(error, `Couldn't import workflow`);
+					this.$router.push({ name: 'NodeViewNew' });
+					return;
+				}
+
+				const nodes = data.workflow.nodes;
+				const hasStartNode = nodes.reduce((accu, node) => accu || node.type === START_NODE_TYPE , false);
```

Comment:
can't we run a find here? it exits early and doesn't iterate over everything
```suggestion
				const hasStartNode = nodes.find((node) => node.type === START_NODE_TYPE)
```

or if you wanna work with booleans:
```suggestion
				const hasStartNode = nodes.find((node) => node.type === START_NODE_TYPE) ? true : false;
```

Reply:
find looks good. thanks for the suggestion.